syncer/client.go

```diff
@@ -210,16 +219,15 @@ func (m *syncPeerClient) handleStatusUpdate(obj interface{}, from peer.ID) {
 		return
 	}
 
-	if atomic.LoadUint64(m.closed) > 0 {
-		m.logger.Debug("received status from peer after client was closed, ignoring", "id", from)
+	m.peerStatusUpdateChLock.Lock()
+	defer m.peerStatusUpdateChLock.Unlock()
 
-		return
-	}
-
-	m.peerStatusUpdateCh <- &NoForkPeer{
-		ID:       from,
-		Number:   status.Number,
-		Distance: m.network.GetPeerDistance(from),
+	if !m.peerStatusUpdateChClosed {
```

Comment:
Sorry for my ignorance in advance, but could we use query `closeCh` state to determine if `Close` function has been called?

Something like this:

```suggestion
    _, isCloseChAlive := <-m.closeCh
	if !isCloseChAlive {
```

If we could do it, then I believe we can get rid of `peerStatusUpdateChLock` and `peerStatusUpdateChClosed`, which would be awesome. WDYT?

Reply:
It wont work because in line 225 it will be alive and in line 226 it wont. I think this is only approach that will work. Its complicated because we are closing channel from totally different routine